app/src/main/kotlin/com/wire/android/util/permission/CallingRecordAudioRequestFlow.kt

```diff
@@ -37,17 +41,25 @@ class CallingAudioRequestFlow(
     private val audioRecordPermissionLauncher: ManagedActivityResultLauncher<Array<String>, Map<String, @JvmSuppressWildcards Boolean>>
 ) {
     fun launch() {
-        if (context.checkPermission(android.Manifest.permission.RECORD_AUDIO) &&
-            context.checkPermission(android.Manifest.permission.BLUETOOTH_CONNECT)
-        ) {
+        val audioPermissionEnabled = context.checkPermission(android.Manifest.permission.RECORD_AUDIO)
+        val bluetoothPermissionEnabled = context.checkPermission(android.Manifest.permission.BLUETOOTH_CONNECT)
+
+        val neededPermissions = arrayOf(
+            android.Manifest.permission.RECORD_AUDIO
+        )
+
+        val permissionsEnabled = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+            neededPermissions[neededPermissions.size] = android.Manifest.permission.BLUETOOTH_CONNECT
+
```

Comment:
This will throw an `OutofBoundException`
To add a new item to the list you may need to use `mutableList` and then convert it to `toTypedArray`

Reply:
Good catch!